### PR TITLE
Cache dependencies

### PR DESCRIPTION
**Pull request by amrita-shrestha**
This PR cache php dependencies
### Related issues
 https://github.com/owncloud/ocis-php-sdk/issues/53

### CODE REVIEW

**Comment by individual-it** [comment]
It seems like every pipeline still downloads all the composer dependencies.

Is this PR also supposed to cache the ocis build, or will that be done in a separate PR?

File: .drone.star [hunk 1]
```diff
@@ -422,3 +443,115 @@ def sonarAnalysis(ctx, phpVersion = DEFAULT_PHP_VERSION):
...
     }]
     return result
+
+def composerInstall():
+    return [{
+        "name": "composer-install",
+        "image": OC_CI_PHP % DEFAULT_PHP_VERSION,
+        "environment": {
+            "COMPOSER_HOME": "%s/.cache/composer" % dir["base"],
+        },
```
Comment (saw-jan) [inline .drone.star:453]: I think we have to use this for all steps that do `composer install` in order for restored cache to work

File: .drone.star [hunk 2]
```diff
@@ -227,17 +253,22 @@ def tests(ctx, tests):
...
                         {
                             "name": test[0],
                             "image": OC_CI_PHP % DEFAULT_PHP_VERSION,
+                            "environment": {
+                                "COMPOSER_HOME": "%s/.cache/composer" % dir["base"],
+                            },
                             "commands": [
-                                "composer install",
+                                "ls -al",
+                                "$COMPOSER_HOME install",
```
Comment (saw-jan) [inline .drone.star:265]: this will probably not work. `composer install` should be fine

**Review by individual-it — CHANGES_REQUESTED** [review]
extra file committed: `ss.zip`

File: Makefile [hunk 3]
```diff
@@ -33,7 +33,7 @@ test-php-integration: run-ocis-with-keycloak
 
 .PHONY: test-php-integration-ci
 test-php-integration-ci:            ## Run php integration tests in CI
-test-php-integration-ci: vendor/bin/phpunit
+	composer install
```
Comment (saw-jan) [inline Makefile:36]: this was fine. or was there any issue with it?
```suggestion
test-php-integration-ci: vendor/bin/phpunit
```

Comment (amrita-shrestha) [reply Makefile:36]: I am trying to figure out what is causing cache not working

File: .drone.star [hunk 4]
```diff
@@ -73,7 +79,7 @@ def main(ctx):
 
 def phpIntegrationTest(ctx, phpversions, coverage):
     pipelines = []
-    steps = buildOcis() + keycloakService() + ocisService()
+    steps = buildOcis() + ocisService() + keycloakService() + cacheRestore()
```
Comment (saw-jan) [inline .drone.star:82]:
```suggestion
    steps = keycloakService() + buildOcis() + ocisService() + cacheRestore()
```

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by individual-it — APPROVED** [review]

**Pull request merged**